modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -538,3 +538,101 @@ RETURN [stop IN nodes(p) | stop.name] AS r1,
+| +Projection                |  1 | (a) ((anon_12)-[anon_14]-(anon_13)-[anon_11]-())* (b) AS p                              |             52 |   45 |       0 |                |                    0/0 |     3.000 |                     |
+| |                          +----+-----------------------------------------------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+                     |
+| +StatefulShortestPath(All) |  2 | SHORTEST 1 GROUPS (a) ((`anon_5`)-[`anon_6`]-(`anon_7`)-[`anon_8`]-(`anon_9`)){1, } (b) |             52 |   45 |     232 |          19687 |                    0/0 |    66.022 | In Pipeline 1       |
+| |                          |    |         expanding from: b                                                               |                |      |         |                |                        |           |                     |
+| |                          |    |     inlined predicates: a:Station                                                       |                |      |         |                |                        |           |                     |
+| |                          +----+-----------------------------------------------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
+| +Filter                    |  3 | b.name IN [$autostring_0, $autostring_1, $autostring_2]                                 |              1 |    3 |      18 |              0 |                        |           |                     |
+| |                          +----+-----------------------------------------------------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
+| +NodeByLabelScan           |  4 | b:Station                                                                               |             10 |    9 |      10 |            376 |                    3/0 |     2.487 | Fused in Pipeline 0 |
++----------------------------+----+-----------------------------------------------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
+----
+
+
+However, the heuristic decision to favor `StatefulShortestPath(All)` can lead to worse query performance.
+To have the planner choose the `StatefulShortestPath(Into)`, rewrite the query using a xref:subqueries/call-subquery.adoc[`CALL` subquery].
+This is because the `CALL` subquery is executed for each incoming row.
+Thus, in the below query, both `a` and `b` are matched to exactly one `Station` node respectively which enables the planner to use `StatefulShortestPath(Into)`.
```

Comment:
It would be worth clarifying that the planner knows that for each invocation of the subquery `CALL`, the planner knows that `a` and `b` are bound to one `Station` node each, and thus the planner uses `StatefulShortestPath(Into)`.

Reply:
I have suggested a change in Jens his PR when this one is merged in